app/models/submission/submission.rb

```diff
@@ -31,7 +31,9 @@ def save_submission!(assignment)
   end
 
   def save_results!(results, assignment)
-    assignment.update! results
+    assignment.failed_submissions_count += 1 unless results[:status] == :passed
+    assignment.update results
+    assignment.save!
```

Comment:
@flbulgarelli, `update` + `save!` is not the same that `update!`?

Reply:
Yes, it is.

Reply:
But only in the scenario you did not assigned unsaved attributes before update.